src/Tizen.TV.UIControls.Forms/Renderer/GridViewRenderer.cs

```diff
@@ -0,0 +1,299 @@
+            };
+        }
+
+        void OnCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
+        {
+            UpdateItemsSource();
+        }
+
+        void UpdateItemsSource()
+        {
+            Control.Clear();
+            _gengridItemDic.Clear();
+            _itemHandleDic.Clear();
+            _itemViewDic.Clear();
+
+            if (_collectionChanged != null)
+            {
+                _collectionChanged.CollectionChanged -= OnCollectionChanged;
+                _collectionChanged = null;
+            }
+
+            if (Element.ItemsSource == null)
+            {
+                return;
+            }
+
+            foreach (var item in Element.ItemsSource)
+            {
+                var context = new GengridItemContext
+                {
+                    Data = item,
+                    RealizedView = null,
+                };
+                var gridItem = Control.Append(_gridItemClass, context);
+                IntPtr handle = gridItem;
+                _itemHandleDic.Add(handle, gridItem);
+                _gengridItemDic.Add(item, gridItem);
+            }
+
+            if (Element.ItemsSource is INotifyCollectionChanged collection)
+            {
+                _collectionChanged = collection;
+                collection.CollectionChanged += OnCollectionChanged;
+            }
+        }
+
+        View CreateOrGetRealView(GengridItemContext context)
+        {
+            if (_itemViewDic.ContainsKey(context.Data))
+            {
+                return _itemViewDic[context.Data].RealizedView;
+            }
+            var realView = CreateView(Element.ItemTemplate);
+            realView.BindingContext = context.Data;
+            context.RealizedView = realView;
+            _itemViewDic.Add(context.Data, context);
+            return realView;
+        }
+
+        View CreateView(DataTemplate template)
+        {
+            if (template != null)
+            {
+                var content = template.CreateContent();
+                if (content is View view)
+                    return view;
+            }
+            return null;
+        }
+
+        void UpdateSelectedItem()
+        {
+            if (Element.SelectedItem == null)
+            {
+                Control.SelectedItem.IsSelected = false;
+                return;
+            }
+            GenGridItem item = null;
+            _gengridItemDic.TryGetValue(Element.SelectedItem, out item);
+            if (item != null)
+            {
+                item.IsSelected = true;
+            }
+        }
+    }
+
+    public static class GridViewConvertExtensions
+    {
+        public static double LayoutAlignmentToDouble(this LayoutAlignment alignment)
+        {
+
+            double alignValue = 0.5;
+            switch (alignment)
+            {
+                case LayoutAlignment.Start:
+                    alignValue = 0.1;
```

Comment:
Why not 0?

Reply:
/// <summary>
        /// Gets or sets the item's grid alignment along X-axis within a given GenGrid widget.
        /// Accepted values are in the 0.0 to 1.0 range, with the special value -1.0 used to specify "justify" or "fill" by some users.
        /// By default, value is 0.0, meaning that the gengrid has its items grid placed exactly in the left along X-axis.
        /// </summary>

Reply:
I fixed it . Thank you.